libkbfs/test_stallers.go

```diff
@@ -38,6 +39,166 @@ const (
 	StallableMDPutUnmerged           StallableMDOp = "PutUnmerged"
 )
 
+type naïveStallContext struct {
+	onStalled <-chan struct{}
+	unstall   chan<- struct{}
+	undoStall func()
+}
+
+// NaïveStaller is used to stall certain ops in BlockOps or MDOps . Unlike
+// StallBlockOp and StallMDOp which provides a way to precisely control which
+// particular op is stalled by passing in ctx with corresponding stallKey,
+// NaïveStaller simply stalls all instances of specified op.
+type NaïveStaller struct {
```

Comment:
I'm not a huge fan, but am also willing to try it out. Note that GitHub's syntax highlighting breaks on it!